quil-rs/src/parser/token.rs

```diff
@@ -67,6 +67,78 @@ where
     }
 }
 
+/// The subset of [`Token`]s which (a) do not contain more specific data and (b) are keywords.  Used
+/// to ensure that keyword-checking remains in sync with the definition of [`Token`].
+#[derive(Debug, Copy, Clone, PartialEq, Eq, strum::Display, strum::EnumString)]
+#[strum(serialize_all = "SCREAMING-KEBAB-CASE")]
+pub enum KeywordToken {
+    As,
+    Matrix,
+    #[strum(serialize = "mut")]
+    Mutable,
+    #[strum(serialize = "NONBLOCKING")]
+    NonBlocking,
+    Offset,
+    PauliSum,
+    Permutation,
+    Sharing,
+}
+
+impl From<KeywordToken> for Token {
+    fn from(token: KeywordToken) -> Self {
+        match token {
+            KeywordToken::As => Token::As,
+            KeywordToken::Matrix => Token::Matrix,
+            KeywordToken::Mutable => Token::Mutable,
+            KeywordToken::NonBlocking => Token::NonBlocking,
+            KeywordToken::Offset => Token::Offset,
+            KeywordToken::PauliSum => Token::PauliSum,
+            KeywordToken::Permutation => Token::Permutation,
+            KeywordToken::Sharing => Token::Sharing,
+        }
+    }
+}
+
+impl TryFrom<Token> for KeywordToken {
+    type Error = ();
+
+    fn try_from(token: Token) -> Result<Self, Self::Error> {
+        // This match is explicit so that if you add a new [`Token`] constructor you have to decide
+        // if it's a keyword.  Please do not add a top-level wildcard match here.
```

Comment:
Would adding `#![deny(clippy::wildcard_enum_match_arm)]` work?

Reply:
Oh perfect, I didn’t know about that!